### ISSUE TITLE
Committing CRLF line separators to the Git repository

### ISSUE THREAD

**Issue by OctaneInteractive** [willdurand/EmailReplyParser#41]
Hi, I was attempting to commit the EmailReplyParser library in Git when I got the following error:

> You are about to commit CRLF line separators to the Git repository.
> It is recommended to set the core.autocrlf Git attribute to input and and avoid line separator issues.

I've done a bit of a Google and I can't find any conclusive advice.

Based on the limited understanding of email that I have, formatting and line endings are crucial, so I'd rather not proceed without some guidance first.

**Comment by willdurand** [comment]
Hi! What's your OS? What do you mean exactly by writing "to commit the EmailReplyParser library in Git"?

**Comment by OctaneInteractive** [comment]
Hi, I'm using OS X 10.9.5 and PHPStorm 10.0.1

<img width="720" alt="screen shot 2015-12-06 at 12 00 47" src="https://cloud.githubusercontent.com/assets/419501/11615485/966edab6-9c59-11e5-9d95-dbbe4dfa3733.png">

**Comment by dkisselev** [comment]
tl;dr: Just click Fix and Commit.

There are two ways to represent a line ending in a source code file, CRLF or just LF. Windows tends to default to CRLF, but the rest of the world uses just LF, so Git/Programming standards in general prefer LF. Nearly ever text editor out there (except for notepad.exe) understands LF just fine, so there's no reason to use CRLF.

Basically PHPStorm is telling you that you're doing something un-ideal, and suggesting to fix it automatically for you.

Chances are that the file you're working on was last edited by someone running Windows so it's in CLRF.

Clicking Fix and Commit will configure Git to just always automatically convert line endings to LF before it commits, and that's a highly recommended/standard thing to do.

**Comment by OctaneInteractive** [comment]
I'd downloaded the files straight from here, which suggests the CLRF characters are in the source code.

**Comment by Veereshdammur** [comment]
@dkisselev Thank you for an explanation.

**Comment by majonathany** [comment]
**Clicking Fix and Commit will configure Git to just always automatically convert line endings to LF before it commits, and that's a highly recommended/standard thing to do.**

Hello, I'd like to revisit whether this is indeed the recommended thing to do (fix and commit). In my experience, if you convert line endings to LF before it commits, this happens for the whole file. This can result in the entire file having a blob difference, like -2015 +2200 line changes. I think this would be great if the entire team is on the same page (better for configuration) - but often developers (especially new developers or young/disorganized teams) don't even know what CRLF/LF are and this can make merge requests more challenging. Often my coworkers use Windows and I use Mac, but other than the line endings, given there's no functional difference between CRLF and LF, it may make users lives more difficult if the line ending is all that changes but colleagues can't diff each line when there inevitably will be merge conflicts.

I think the line endings should be left as-is, and Git disregard any differences between the /n and /r/n tokens, this way it abstracts away that from the user and git can produce clearer diffs when developers inspect commits or are solving merge conflicts.
